sandbox/libs/authn/src/main/java/org/opensearch/authn/User.java

```diff
@@ -18,23 +18,23 @@
 
 public class User {
 
-    @JsonProperty(value = "primary_principal")
-    private StringPrincipal primaryPrincipal;
+    @JsonProperty(value = "username")
+    private StringPrincipal username;
 
     @JsonProperty(value = "hash")
     private String bcryptHash;
 
     @JsonProperty(value = "attributes")
     private Map<String, String> attributes = Collections.emptyMap();
 
-    @JsonProperty(value = "primary_principal")
-    public StringPrincipal getPrimaryPrincipal() {
-        return primaryPrincipal;
+    @JsonProperty(value = "username")
```

Comment:
Why this change, are we renaming Principal terminology or are we reserving it for something else?

Reply:
I think we should use `username` for the primary principal as a convention. Do you think this should be changed back? I think using username makes the work in identity more readable as username is currently the primary identifier for a subject in OpenSearch.

Reply:
it should work for now, but this should definitely be decided upon as to what naming are we proceeding with